jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -8325,13 +8325,139 @@ public String indent(int n) {
 /*[ENDIF] Java12 */
 
 /*[IF Java13]*/
-	/*
+	/**
 	 * Determine if current String object is LATIN1.
 	 *
 	 * @return true if it is LATIN1, otherwise false.
 	 */
 	boolean isLatin1() {
 		return LATIN1 == coder();
 	}
+
+	/**
+	 * Format the string using this string as the format with supplied args.
+	 *
+	 * @param args
+	 *          the format arguments to use
+	 *
+	 * @return the formatted result
+	 *
+	 * @see #format(String, Object...)
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String formatted(Object... args) {
+		return String.format(this, args);
+	}
+
+	/**
+	 * Removes the minimum indentation from the beginning of each line and
+	 * removes the trailing spaces in every line from the string
+	 *
+	 * @return this string with incidental whitespaces removed from every line
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String stripIndent() {
+		Iterator<String> iter = lines().iterator();
+		StringBuilder builder = new StringBuilder();
+		int min = Integer.MAX_VALUE;
+
+		while (iter.hasNext()) {
+			String line = iter.next();
+
+			if ((!line.isBlank()) || (!iter.hasNext())) {
+				int count = 0;
+				while ((count < line.length())
+					&& (Character.isWhitespace(line.charAt(count)))
+				) {
+					count++;
+				}
+				min = (count < min ? count : min);
+			}
+
+			if (line.isBlank()) {
+				builder.append("");
+			} else {
+				builder.append(line.stripTrailing());
+			}
+			if (iter.hasNext()) {
+				builder.append("\n");
+			}
+		}
+
+		return builder.toString().indent(-min);
+	}
+
+	/**
+	 * Translate the escape sequences in this string as if in a string literal
+	 *
+	 * @return result string after translation
+	 *
+	 * @throws IllegalArgumentException
+	 *          If invalid escape sequence is detected
+	 */
+	@Deprecated(forRemoval=true, since="13")
+	public String translateEscapes() {
+		StringBuilder builder = new StringBuilder();
+		char[] charArray = toCharArray();
+		int index = 0;
+		while (index < length()) {
+			if (charArray[index] == '\\') {
+				index++;
+				int octal = 0;
+				switch (charArray[index]) {
+					case 'b':
+						builder.append('\b');
+						break;
+					case 't':
+						builder.append('\t');
+						break;
+					case 'n':
+						builder.append('\n');
+						break;
+					case 'f':
+						builder.append('\f');
+						break;
+					case 'r':
+						builder.append('\r');
+						break;
+					case '\"':
+					case '\'':
+					case '\\':
+						builder.append(charArray[index]);
+						break;
+					case '0':
+					case '1':
+					case '2':
+					case '3':
+						octal = charArray[index] - '0';
+						if ((index + 1 < length()) && ('0' <= charArray[index + 1]) && ('7' >= charArray[index + 1])) {
```

Comment:
`index + 1 < length()` will fail if length is `Integer.MAX_VALUE`: use `index < length() - 1` instead (here and below).

I think a local capturing `length()` is warranted in this method.

Reply:
fixed, thanks